File: examples/typegraphs/cors.py
# Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
# SPDX-License-Identifier: MPL-2.0

# skip:start
from typegraph import Graph, Policy, t, typegraph
from typegraph.graph.params import Cors
from typegraph.runtimes.random import RandomRuntime


# skip:end
@typegraph(
    # highlight-start
    cors=Cors(
        allow_origin=["https://not-this.domain"],
        allow_headers=["x-custom-header"],
        expose_headers=["header-1"],
        allow_credentials=True,
        max_age_sec=60,
    ),
    # highlight-end
)
def cors(g: Graph):
    random = RandomRuntime(seed=0, reset=None)

    g.expose(
        Policy.public(),
        catch_me_if_you_can=random.gen(t.string()),
    )

File: examples/typegraphs/cors.ts
// skip:start
import { Policy, t, typegraph } from "@typegraph/sdk/index.ts";
import { RandomRuntime } from "@typegraph/sdk/runtimes/random.ts";

// skip:end

await typegraph(
  {
    name: "cors",
    // highlight-start
    cors: {
      allowOrigin: ["https://not-this.domain"],
      allowHeaders: ["x-custom-header"],
      exposeHeaders: ["header-1"],
      allowCredentials: true,
      maxAgeSec: 60,
    },
    // highlight-end
  },
  (g) => {
    const random = new RandomRuntime({ seed: 0 });

    g.expose(
      {
        catch_me_if_you_can: random.gen(t.string()),
      },
      Policy.public()
    );
  }
);
